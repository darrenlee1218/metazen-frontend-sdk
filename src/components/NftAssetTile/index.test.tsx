import React from 'react';
import { render, fireEvent, screen, cleanup } from '@testing-library/react';
import { ComponentProps } from 'react';
import { AssetVariant, NftAssetTile } from '.';
import { TestProviders } from '@utils/testing/TestProviders';

afterEach(cleanup);

const defaultProps: ComponentProps<typeof NftAssetTile> = {
  variant: AssetVariant.Collection,
  assetIconUrl: 'assetIcon.jpg',
  networkIconUrl: undefined,
  assetName: 'NFT Asset',
  assetQuantity: '1',
  onClick: () => {},
};

it('calls the onClick callback when there is a click event', () => {
  const callback = jest.fn();
  const props = { ...defaultProps, onClick: callback };
  render(<NftAssetTile {...props} />, { wrapper: TestProviders });
  fireEvent.click(screen.getByAltText('NFT Asset image'));
  expect(callback).toHaveBeenCalled();
});

it('renders the quantity badge within the image when variant is Item', () => {
  const props = { ...defaultProps, variant: AssetVariant.Item, assetQuantity: '100' };
  render(<NftAssetTile {...props} />, { wrapper: TestProviders });
  const imageElement = screen.getByAltText('NFT Asset image');
  expect(imageElement.nextSibling).toHaveTextContent('100');
  const descriptorElement = screen.getByTestId('asset-descriptor');
  expect(descriptorElement.textContent).toBe('NFT Asset');
});

it('does not render quantity badge within image when assetQuantity is 1', () => {
  const props = { ...defaultProps, variant: AssetVariant.Item, assetQuantity: '1' };
  render(<NftAssetTile {...props} />, { wrapper: TestProviders });
  const imageElement = screen.getByAltText('NFT Asset image');
  expect(imageElement.nextSibling).toBe(null);
});

it('renders the quantity text in the descriptor when variant is Collection', () => {
  const props = { ...defaultProps, variant: AssetVariant.Collection, assetQuantity: '100' };
  render(<NftAssetTile {...props} />, { wrapper: TestProviders });
  const imageElement = screen.getByAltText('NFT Asset image');
  expect(imageElement.nextSibling).not.toBeInTheDocument();
  const descriptorElement = screen.getByTestId('asset-descriptor');
  expect(descriptorElement.textContent).toBe(['NFT Asset', '100 items'].join(''));
});

it('renders a skeleton when variant is Collection but assetQuantity is undefined', () => {
  const props = { ...defaultProps, variant: AssetVariant.Collection, assetQuantity: undefined };
  render(<NftAssetTile {...props} />, { wrapper: TestProviders });
  expect(screen.getByTestId('asset-descriptor')).toMatchSnapshot();
});

it('renders the networkIcon when it is defined', () => {
  const props = { ...defaultProps, networkIconUrl: 'networkIcon.jpg' };
  render(<NftAssetTile {...props} />, { wrapper: TestProviders });
  const imageElement = screen.getByAltText('network icon');
  expect(imageElement).toBeInTheDocument();
});

it('matches snapshot', () => {
  const { container } = render(<NftAssetTile {...defaultProps} />, { wrapper: TestProviders });
  expect(container).toMatchSnapshot();
});
