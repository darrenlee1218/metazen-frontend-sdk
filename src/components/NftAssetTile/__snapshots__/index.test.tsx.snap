// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root MuiBox-root css-1lqyjo8-MuiButtonBase-root-MuiCardActionArea-root"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiBox-root css-hjbv5f"
    >
      <img
        alt="NFT Asset image"
        class="MuiBox-root css-djl8wu"
        src="assetIcon.jpg"
      />
    </div>
    <div
      class="MuiBox-root css-120alf"
      data-testid="asset-descriptor"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-11hf9u5-MuiTypography-root"
      >
        NFT Asset
      </h5>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1xin5va-MuiTypography-root"
      >
        1
         items
      </h5>
    </div>
    <span
      class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;

exports[`renders a skeleton when variant is Collection but assetQuantity is undefined 1`] = `
<div
  class="MuiBox-root css-120alf"
  data-testid="asset-descriptor"
>
  <h5
    class="MuiTypography-root MuiTypography-h5 css-11hf9u5-MuiTypography-root"
  >
    NFT Asset
  </h5>
  <h5
    class="MuiTypography-root MuiTypography-h5 css-1xin5va-MuiTypography-root"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1h2g3k5-MuiSkeleton-root"
      style="width: 12px;"
    />
  </h5>
</div>
`;
